sensor/kubernetes/clusterstatus/updater.go

```diff
@@ -313,3 +328,59 @@ func NewUpdater(client client.Interface) common.SensorComponent {
 		getProviders: providers.GetMetadata,
 	}
 }
+
+func (u *updaterImpl) getMetadataFromInfrastructureCR(ctx context.Context) (*storage.ProviderMetadata, error) {
+	clusterInfraCR, err := u.client.OpenshiftConfig().ConfigV1().
+		Infrastructures().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return providerMetadataFromInfraCR(clusterInfraCR), nil
+}
+
+func providerMetadataFromInfraCR(infra *configv1.Infrastructure) *storage.ProviderMetadata {
+	// The platform status is required to read out the provider specific information. If it is unset,
+	// we can short-circuit here.
+	if infra.Status.PlatformStatus == nil {
+		return nil
+	}
+
+	switch infra.Status.PlatformStatus.Type {
+	case configv1.AWSPlatformType:
+		return &storage.ProviderMetadata{
+			Region:   infra.Status.PlatformStatus.AWS.Region,
+			Provider: &storage.ProviderMetadata_Aws{Aws: &storage.AWSProviderMetadata{}},
+			Verified: true,
+			Cluster: &storage.ClusterMetadata{
+				Type: storage.ClusterMetadata_OCP,
```

Comment:
This would set the incorrect type for OSD/ROSA clusters. We can get the correct type from `platformStatus.aws.resourceTags`. See for example:

```
  platformStatus:
    aws:
      region: us-east-1
      resourceTags:
        - key: red-hat-managed
          value: 'true'
        - key: red-hat-clustertype
          value: osd  # or rosa
```